### PR TITLE
Cleanups in Specs

### PR DESCRIPTION
**Pull request by jnyrup**
Some cleanups in Specs. 
A little dogfooding and getting rid of a distraction method. 

I haven't heard it used before, so I'll pretend I coined that term for "helper methods" that are unhelpful and simply a distraction.

### CODE REVIEW

File: Tests/FluentAssertions.Specs/Exceptions/MiscellaneousExceptionSpecs.cs [hunk 1]
```diff
@@ -62,52 +62,33 @@ private static IEnumerable<char> MethodThatUsesYield(string bar)
...
-            {
-                // Assert
-                exc.Message.Should().StartWith(
-                    "Expected exception where (e.Message == \"Error\"), but the condition was not met");
-            }
-            catch (Exception exc)
-            {
-                exc.Message.Should().StartWith(
-                    "Expected exception where (e.Message == \"Error\"), but the condition was not met");
-            }
```
Comment (jnyrup) [inline Tests/FluentAssertions.Specs/Exceptions/MiscellaneousExceptionSpecs.cs:110]: I have no idea why this was implemented with two `catch`es...

File: Tests/FluentAssertions.Specs/Exceptions/InnerExceptionSpecs.cs [hunk 2]
```diff
@@ -165,77 +139,53 @@ public void WithInnerExceptionExactly_when_subject_throws_expected_inner_excepti
                     .WithInnerExceptionExactly<ArgumentNullException>("because {0} should do just that", "the action");
         }
 
-        private static string BuildExpectedMessageForWithInnerExceptionExactly(string because, string innerExceptionMessage)
```
Comment (jnyrup) [inline Tests/FluentAssertions.Specs/Exceptions/InnerExceptionSpecs.cs:168]: One ~helper~ distractor method less 🔥

**Comment by dennisdoomen** [comment]
The reason why I used the old-style `try...catch` construct is to avoid false negatives because a bug in `Should().Throw()` may get hidden by itself.

**Comment by jnyrup** [comment]
> The reason why I used the old-style `try...catch` construct is to avoid false negatives because a bug in `Should().Throw()` may get hidden by itself.

Ahh of course.

I'll revert the ones that had `throw new XunitException("This point should not be reached.")`

**Comment by dennisdoomen** [comment]
I guess it is a bit hard which require explicit `try...catch` and which ones are fine to be dogfooding using `Should().Throw()`.

**Review by dennisdoomen — APPROVED** [review]

**Comment by dennisdoomen** [comment]
You are having second thoughts @jnyrup? Or did you lost track of your own PR? 😜

**Comment by jnyrup** [comment]
Partly on both. Been more busy on others PRs.

**Pull request merged**